js/gdm/authPrompt.js

```diff
@@ -183,10 +229,17 @@ const AuthPrompt = new Lang.Class({
 
         this._entry.clutter_text.connect('text-changed',
                                          Lang.bind(this, function() {
-                                             if (!this._userVerifier.hasPendingMessages)
-                                                 this._fadeOutMessage();
-
-                                             this._updateNextButtonSensitivity(this._entry.text.length > 0);
+                                             if (this._passwordResetCode == null &&
+                                                 !this._displayingPasswordHint) {
```

Comment:
Hm, this needs to be moved to the next conditional down, or entering the correct password after requesting a hint won't work. I'll fix it.